app/models/post.rb

```diff
@@ -1,11 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :user, foreign_key: 'user_id'
   has_many :comments
   has_many :likes
   after_create :update_user_posts_counter
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   def update_user_posts_counter
-    author.update(posts_counter: author.posts.count)
+    user.update(posts_counter: user.posts.count)
+    # user.reload
```

Comment:
- Let's keep our previous code as it follows the ERD structure for the database.